docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -0,0 +1,170 @@
+[[synthetic-source]]
+==== Synthetic `_source`
+
+Though very handy to have around, the source field takes up a significant amount
+of space on disk. Instead of storing source documents on disk exactly as you
+send them, Elasticsearch can reconstruct source content on the fly. Enable this
+by setting `synthetic: true` in `_source`:
+
+[source,console,id=enable-synthetic-source-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": {
+      "synthetic": true
+    }
+  }
+}
+----
+// TESTSETUP
+
+While this on the fly reconstruction is *generally* slower than saving the source
+documents verbatim and loading them at query time, it saves a lot of storage
+space. There are a couple of restrictions to be aware of:
+
+* When you retrieve sythetic `_source` content it undergoes minor
+<<synthetic-source-modifications,modifications>> compared to the original JSON.
+* Synthetic `_source` can be used with indices that contain only these field
+types:
+
+** <<boolean-synthetic-source,`boolean`>>
+** <<numeric-synthetic-source,`byte`>>
+** <<numeric-synthetic-source,`double`>>
+** <<numeric-synthetic-source,`float`>>
+** <<geo-point-synthetic-source,`geo_point`>>
+** <<numeric-synthetic-source,`half_float`>>
+** <<numeric-synthetic-source,`integer`>>
+** <<ip-synthetic-source,`ip`>>
+** <<keyword-synthetic-source,`keyword`>>
+** <<numeric-synthetic-source,`long`>>
+** <<numeric-synthetic-source,`scaled_float`>>
+** <<numeric-synthetic-source,`short`>>
+** <<text-synthetic-source,`text`>>
```

Comment:
isn't this half the truth given that text has to have a keyword sub-field?

Reply:
The link says that, yeah. Do you think I should add it in the list?

Reply:
Got it, maybe add (with a keyword sub-field) in the list?

Reply:
:+1:

Reply:
5962c2716ea